src/Bramus/Router/Router.php

```diff
@@ -42,10 +47,56 @@ class Router
      */
     private $serverBasePath;
 
-    /**
-     * @var string Default Controllers Namespace
-     */
-    private $namespace = '';
+	/**
+	 * @var string Default Controllers Namespace
+	 */
+	private $namespace = '';
+
+	/**
+	 * @var string The method that will take precedent over any auto determined request method
+	 */
+	private $requestMethodOverride = '';
+
+	/**
+	 * @var string The URI that will take precedent over any auto determined request URI
+	 */
+	private $requestUriOverride = '';
+
+	/**
+	 * @var bool To indicate that the URI has been requested to be overwritten, in case of an empty request URI
+	 */
+	private $requestUriOverridden = false;
+
+	/**
+	 * Allow the method of the current request to be overridden instead of being derived from the $_SERVER environment global
+	 *
+	 * @see getRequestMethodCandidate()
+	 *
+	 * @param $method The HTTP method to handle this request with
+	 *
+	 * @throws \Exception with invalid HTTP method
+	 */
+	public function setRequestMethodOverride($method)
+	{
+		$method = trim(strtoupper($method));
+
+		if(!in_array($method, explode('|', self::HTTP_METHODS))) throw new \Exception('Invalid HTTP method', 405);
+
+		$this->requestMethodOverride = $method;
+	}
+
+	/**
+	 * Allow the method of the current request to be overridden instead of being derived from the $_SERVER environment global
+	 *
+	 * @see getRequestUriCandidate()
+	 *
+	 * @param $requestUri The request URI to handle this request with
+	 */
+	public function setRequestUriOverride($requestUri)
```

Comment:
It would be great to add some proper variable filtering here. This code may not be XSS bullet proof... Of course this would be a problem of `user-side` because there are plenty of ways to passing the `$requestUri` argument. 

Of course if the URI doesn't exists the router will just ignore, and we have also have a clean environment for the variable extraction `ob_*`...

Still...  I really feel that we should check if the given REQUEST URI is valid and doesn't point to a different host. And doesn't contain injection (would be nice).

Reply:
- [ ] maybe filter with `FILTER_VALIDATE_URL` + `filter_var()`